app/models/manageiq/providers/hawkular/inventory/persister/availability_updates.rb

```diff
@@ -1,40 +1,44 @@
 module ManageIQ::Providers
   class Hawkular::Inventory::Persister::AvailabilityUpdates < Hawkular::Inventory::Persister::MiddlewareManager
     def self.save_deployments(ems, collection)
-      ::ActiveRecord::Base.transaction do
-        collection.to_a.each do |item|
-          deployment = ems.middleware_deployments.find_by(:ems_ref => item.manager_uuid)
-          next unless deployment # if deployment is not found in the database, it is ignored.
+      save_resource(ems, collection, :deployment) do |deployment, item|
+        deployment.status = item.status
+      end
+    end
 
-          $mw_log.debug("EMS_#{ems.id}(Persister::AvailabilityUpdates): " \
-                        "Updating status #{deployment.status} -> #{item.status} for deployment #{deployment.ems_ref}")
+    def self.save_servers(ems, collection)
+      save_resource(ems, collection, :server) do |server, item|
+        data_to_update = item.properties.try(:slice, 'Server State', 'Availability', 'Calculated Server State')
+        next if data_to_update.blank?
 
-          deployment.status = item.status
-          deployment.save!
-        end
+        server.properties = {} if server.properties.blank?
```

Comment:
This is code from [previous implementation](https://github.com/ManageIQ/manageiq-providers-hawkular/pull/67/files/e9242b6854fa54b6a0b385fdffa2ded4956a4684#diff-a41892d210b721576656574d8d9d0846L30). I would leave it in this PR and change it in different one.

I think this LOC is a code smell that comes from the definition of `MiddlewareServer#properties` which is exactly done as you said. I think it would be differently implemented because there is no assignment in that. Then when super is called, the new Hash is unbound to the actual MWServer instance.
I am adding a new pull request with that, right after this one will be merged.